core/src/main/java/org/elasticsearch/cluster/metadata/MetaDataCreateIndexService.java

```diff
@@ -398,7 +398,9 @@ public ClusterState execute(ClusterState currentState) throws Exception {
                     if (request.state() == State.OPEN) {
                         RoutingTable.Builder routingTableBuilder = RoutingTable.builder(updatedState.routingTable())
                                 .addAsNew(updatedState.metaData().index(request.index()));
-                        RoutingAllocation.Result routingResult = allocationService.reroute(ClusterState.builder(updatedState).routingTable(routingTableBuilder.build()).build());
+                        RoutingAllocation.Result routingResult = allocationService.reroute(
+                                ClusterState.builder(updatedState).routingTable(routingTableBuilder.build()).build(),
+                                "reroute after creating index [" + request.index() + "]");
```

Comment:
we should try to keep these concise - can we do `"index [" + .. + "] created"` ?